core/modules/util/CmdParser.h

```diff
@@ -0,0 +1,291 @@
+/*
+ * LSST Data Management System
+ * Copyright 2018 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_UTIL_CMD_PARSER_H
+#define LSST_QSERV_UTIL_CMD_PARSER_H
+
+/// CmdParser.h declares:
+///
+/// class CmdParser
+/// (see individual class documentation for more information)
+
+// System headers
+
+#include <map>
+#include <ostream>
+#include <set>
+#include <string>
+#include <vector>
+
+// Qserv headers
+
+// This header declarations
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+/**
+ * The command line parser class for the command-line applications.
+ * It helps with parsing and interpreting command line arguments into
+ * positional parameters, flags and options:
+ *
+ *   <parameter>
+ *   --<flag>
+ *   --<option>=<value>
+ *
+ * Parameters, flags and options can interleave in any order:
+ *
+ *   --flag1 <pos1> --opt1=val1 <pos2> <pos3> ...
+ *
+ * The constructor will throw exception std::invalid_argument if non-conforming
+ * arguments will be found in the command line, such as:
+ *
+ *   --
+ *   --<option>=
+ */
+class CmdParser {
+
+public:
+
+    /**
+     * Return 'true' if the specified string is found in the collection.
+     *
+     * @param val - a string to be evaluated
+     * @param col - a collection
+     */
+    static bool found_in (std::string const&              val,
```

Comment:
I couldn't find any such requirements for function or variable names in the LSST C++ Style Guide. The Guide requires using camel case for namespaces (3-6) and the 'typedef' definitions (3-5). Besides the protobuf methods already use underscores. And so does C++ Standard library and Boost.

Reply:
So, obviously it's a matter of preference. In some cases I find underscore a better option than camel case because it results in a code which easier to parse by humans. A downside is - longer names.

Reply:
I ended p renaming this function into just "in", which makes a sense in this case